### ISSUE TITLE
Video HLS bug

### ISSUE THREAD

**Issue by para2022** [WWBN/AVideo#8850]
The plugin is crashing my browser making it hang when i view a certain channel, when it deactivate it then i can view that channel and the site doesnt hang. and i see errors in my log

[Fri Feb 16 16:35:36.928033 2024] Got error 'sts file SCRIPT_NAME: /view/channel.php; PHP message: AVideoLog::WARNING: VideoHLS:getSourceFile:returnAllSources Your remote file take too long to respond (https://ourtube.co.uk/videos/video_220729084705_va3f1/index.m3u8?token=eyJzYWx0IjoiaHR0cHM6XC9cL291cnR1YmUuY28udWtcLyIsInRva2VuVGltZU91dCI6IjIxNjAwIiwiZmlsZW5hbWUiOiJ2aWRlb18yMjA3MjkwODQ3MDVfdmEzZjEiLCJ0aW1lIjoxNzA4MTAxMzMyfQ==), you may want to use your internal network to communicate. so you can add this domain in your hosts file SCRIPT_NAME: /view/channel.php; PHP message: AVideoLog::ERROR: Time: 35.186s  | Limit: 0.7s | Location: /home/ourtube/public_html/ourtube/view/channel.php Line 450 [/home/ourtube/public_html/ourtube/view/channelBody.php - channelName: AnthonyNesci] IP=2a02:c7c:5f98:6500:3c0f:6e97:f8f7:5192 backtrace=[{"file":"\\/home\\/ourtube\\/public_html\\/ourtube\\/view\\/channelBody.php","line":450,"function":"TimeLogEnd","args":["\\/home\\/ourtube\\/public_html\\/ourtube\\/view\\/channelBody.php - channelName: AnthonyNesci",450]},{"file":"\\/home\\/ourtube\\/public_html\\/ourtube\\/view\\/channel.php","line":41,"args":["\\/home\\/ourtube\\/public_html\\/ourtube\\/view\\/channelBody.php"],"function":"include"}] SCRIPT_NAME: /view/channel.php; PHP message: AVideoLog::DEBUG: session_start takes 1.9706950187683 seconds to open SCRIPT_NAME: /view/channel.php; PHP message: AVideoLog::DEBUG: [{"file":"\\/home\\/ourtube\\/public_html\\/ourtube\\/objects\\/user.php","line":1310,"function":"_session_start","args":[]},{"file":"\\/home\\/ourtube\\/public_html\\/ourtube\\/objects\\/user.php","line":1348,"function":"recreateLoginFromCookie","class":"User","type":"::","args":[]},{"file":"\\/home\\/ourtube\\/public_html\\/ourtube\\/objects\\/Object.php","line":852,"function":"isLogged","class":"User","type":"::","args":[]},{"file":"\\/home\\/ourtube\\/public_html\\/ourtube\\/objects\\/Object.php","line":879,"function":"getCacheDir","class":"ObjectYPT","type":"::","args":["e17de3e09732c98f00043e946b5e157953b13479",true,true,false]},{"file":"\\/home\\/ourtube\\/public_html\\/ourtube\\/objects\\/Object.php","line":568,"function":"getCacheFileName","class":"ObjectYPT","type":"::","args":["getEncoderURL",true,true,false]},{"file":"\\/home\\/ourtube\\/public_html\\/ourtube\\/objects\\/configuration.php","line":569,"function":"setCache","class":"ObjectYPT","type":"::","args":["getEncoderURL","https:\\/\\/en...; PHP message: AVideoLog::DEBUG: cache logTime: 569 getEncoderURL 1.971 ["Finish "] SCRIPT_NAME: /view/channel.php; PHP message: AVideoLog::DEBUG: cache logTime: 573 getEncoderURL 1.971 ["Finish "] SCRIPT_NAME: /view/channel.php; PHP message: AVideoLog::DEBUG: cache logTime: 575 getEncoderURL 1.971 ["Finish "] SCRIPT_NAME: /view/channel.php; PHP message: AVideoLog::WARNING: Time: 1.997s   | Limit: 0.2s | Location: /home/ourtube/public_html/ourtube/view/channel.php Line 137 [navbar.php] IP=2a02:c7c:5f98:6500:3c0f:6e97:f8f7:5192 SCRIPT_NAME: /view/channel.php', referer: https://ourtube.co.uk/

**Comment by para2022** [comment]
This is the channel that hangs my browser:

https://ourtube.co.uk/channel/AnthonyNesci

He has been uploading today so their is nothing wrong with his channel.

**Comment by para2022** [comment]
I have reinstalled the plugin and ive been into his channel and he has 4 HLS videos that all play. Still cannot load his channel.

**Comment by DanielnetoDotCom** [comment]
`VideoHLS:getSourceFile:returnAllSources Your remote file takes too long to respond`

it is just a warning, I recommend you add the ourtube.co.uk to your hosts file

`127.0.0.1 ourtube.co.uk `

The page loads fine for me

![image](https://github.com/WWBN/AVideo/assets/18009293/0839cf04-24e0-4d5d-aff0-8952f00c6ff2)

**Comment by para2022** [comment]
Yes it loads because i just said ive turned off the plugin because its hanging the site. And it is already added in my hosts file.

**Comment by DanielnetoDotCom** [comment]
from your server make a `wget https://ourtube.co.uk/videos/video_220729084705_va3f1/index.m3u8?token=eyJzYWx0IjoiaHR0cHM6XC9cL291cnR1YmUuY28udWtcLyIsInRva2VuVGltZU91dCI6IjIxNjAwIiwiZmlsZW5hbWUiOiJ2aWRlb18yMjA3MjkwODQ3MDVfdmEzZjEiLCJ0aW1lIjoxNzA4MTAxMzMyfQ==`

check if it works fine, if it does try a php script, create it manually

### Using cURL

```php
<?php

$url = 'https://ourtube.co.uk/videos/video_220729084705_va3f1/index.m3u8?token=eyJzYWx0IjoiaHR0cHM6XC9cL291cnR1YmUuY28udWtcLyIsInRva2VuVGltZU91dCI6IjIxNjAwIiwiZmlsZW5hbWUiOiJ2aWRlb18yMjA3MjkwODQ3MDVfdmEzZjEiLCJ0aW1lIjoxNzA4MTAxMzMyfQ==';

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
$content = curl_exec($ch);
curl_close($ch);

if ($content !== false) {
    echo nl2br(htmlspecialchars($content));
} else {
    echo "Failed to download the playlist.";
}

?>
```

### Using file_get_contents

```php
<?php

$url = 'https://ourtube.co.uk/videos/video_220729084705_va3f1/index.m3u8?token=eyJzYWx0IjoiaHR0cHM6XC9cL291cnR1YmUuY28udWtcLyIsInRva2VuVGltZU91dCI6IjIxNjAwIiwiZmlsZW5hbWUiOiJ2aWRlb18yMjA3MjkwODQ3MDVfdmEzZjEiLCJ0aW1lIjoxNzA4MTAxMzMyfQ==';

// Using file_get_contents to fetch the playlist content
$options = [
    "http" => [
        "method" => "GET",
        "header" => "User-Agent: PHP"
    ]
];
$context = stream_context_create($options);
$content = file_get_contents($url, false, $context);

if ($content !== false) {
    echo nl2br(htmlspecialchars($content));
} else {
    echo "Failed to download the playlist.";
}

?>
```

**Comment by para2022** [comment]
ok that was weird. HLS never promoted me to update in the first place. I updated it anyway like i said. I didnt have cache turned on. I restarted apache and went back to plugins and it promoted me to update from 10.0 to 10.2 and now its working fine with HLS turned back on and now i have no errors so problem is resolved. Thanks

https://ourtube.co.uk/channel/AnthonyNesci
